### PR TITLE
Re-enable new calculation functions

### PR DESCRIPTION
**Pull request by nex3**
See https://github.com/sass/sass/issues/3504
See https://github.com/sass/sass-spec/pull/1934
See https://github.com/sass/dart-sass/issues/1934

### CODE REVIEW

**Comment by nex3** [comment]
I accidentally hit merge early on #2076, so I had to recreate it. Marking as a draft until I finish updating it to match sass/sass#3682.

File: lib/src/visitor/async_evaluate.dart [hunk 1]
```diff
@@ -2695,6 +2669,37 @@ final class _EvaluateVisitor
...
       };
 
+  /// Throws an error if [elements] contains two adjacent non-string values.
+  void _checkAdjacentCalculationValues(
+      List<Object> elements, ListExpression node) {
+    assert(elements.length > 1);
+
+    for (var i = 1; i < elements.length; i++) {
+      var previous = elements[i - 1];
+      var next = elements[i];
```
Comment (Goodwine) [inline lib/src/visitor/async_evaluate.dart:2679]: nit: `current` instead of `next`

**Review by Goodwine — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,31 @@
...
+  than before. However, all interpolations that would have produced valid CSS
+  will continue to work, so this is not considered a breaking change.
+
+* Interpolations in calculation functions that aren't used in a position that
+  could also have a normal calculation value are now deprecated. For example,
+  `calc(1px #{"+ 2px"})` is deprecated, but `calc(1px + #{"2px"})` is still
+  allowed. This deprecation is named `calc-interp`. See [the Sass website] for
+  more information.
+
+  [the Sass website]: https://sass-lang.com/d/calc-interp
```
Comment (stof) [inline CHANGELOG.md:27]: This page does not exist on the website.

Comment (nex3) [reply CHANGELOG.md:27]: Oops, we decided not to do that deprecation but never updated the changelog.

**Pull request merged**